ui/src/dashboards/utils/protoboardSuggestion.ts

```diff
@@ -0,0 +1,164 @@
+// Libraries
+import _ from 'lodash'
+// Utils
+import {proxy} from 'src/utils/queryUrlGenerator'
+// Types
+import {Source, Protoboard} from 'src/types'
+
+interface SeriesObject {
+  measurement: string
+  host: string
+}
+
+export const getSuggestedProtoboards = async (
+  source: Source,
+  protoboards: Protoboard[]
+) => {
+  const hosts = await getHosts(source)
+
+  if (!hosts) {
+    return []
+  }
+
+  const newHosts = await addAppsToHosts(source, hosts, protoboards)
+
+  const suggestedProtoboards = _.reduce(
+    newHosts,
+    (acc, host) => {
+      if (host.apps && host.apps.length > 0) {
+        return _.union(acc, host.apps)
+      }
+      return acc
+    },
+    []
+  )
+
+  return suggestedProtoboards
+}
+
+export const addAppsToHosts = async (
+  source: Source,
+  hosts,
+  protoboards: Protoboard[]
+) => {
+  const measurementsList = []
+  const measurementsToProtoboards = {}
+
+  _.forEach(protoboards, pb => {
+    _.forEach(pb.meta.measurements, m => {
+      measurementsToProtoboards[m] = pb.meta.name
+      measurementsList.push(m)
+    })
+  })
+
+  const joinedMeasurements = measurementsList.join('|')
+
+  const newHosts = {...hosts}
+
+  const allSeries = await getAllSeries(source, joinedMeasurements)
+
+  allSeries.forEach(series => {
+    const {measurement, host} = parseSeries(series)
+    if (!newHosts[host]) {
+      return
+    }
+
+    const existingApps = newHosts[host].apps || []
+
+    newHosts[host].apps = _.uniq([
+      ...existingApps,
+      measurementsToProtoboards[measurement],
+    ])
+  })
+
+  return newHosts
+}
+
+export const getHosts = async source => {
+  const hosts = {}
+
+  const query = `SHOW TAG VALUES WITH KEY = "host" WHERE TIME > now() - 10m;`
```

Comment:
how did you decide on `TIME > now() - 10m` as the time range?

Reply:
>"we just want the latest shard" 
> _-@nhaugo_